### PR TITLE
Use VecDeque for Decompositions.

### PR DESCRIPTION
**Pull request by clarcharr**
This optimises `remove(0)` to be `pop_front()` instead. This optimisation is used by the unicode-normalization crate but not this crate.

### CODE REVIEW

File: unic/normal/src/decompose.rs [hunk 1]
```diff
@@ -80,11 +81,11 @@ impl<I: Iterator<Item = char>> Iterator for Decompositions<I> {
         match self.buffer.first() {
```
Comment (CAD97) [inline unic/normal/src/decompose.rs]: `Vec::first` -> `VecDeque::front`

**Comment by CAD97** [comment]
Oh and also, thanks! A new pair of eyes spotting always helps out!

**Comment by behnam** [comment]
Thanks, @clarcharr, for the PR. Looks great, just need to get all the checks green before we can land.

**Comment by clarcharr** [comment]
It's passing now!

**Pull request merged**